server/src/main/java/io/crate/replication/logical/RemoteClusterStateMappingTracker.java

```diff
@@ -0,0 +1,200 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.replication.logical;
+
+import io.crate.common.annotations.VisibleForTesting;
+import io.crate.common.unit.TimeValue;
+import io.crate.metadata.RelationName;
+import io.crate.replication.logical.metadata.Publication;
+import io.crate.replication.logical.metadata.PublicationsMetadata;
+import io.crate.replication.logical.metadata.SubscriptionsMetadata;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.admin.cluster.state.ClusterStateAction;
+import org.elasticsearch.action.admin.cluster.state.ClusterStateResponse;
+import org.elasticsearch.action.bulk.BackoffPolicy;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.ClusterStateUpdateTask;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.logging.Loggers;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.concurrent.EsRejectedExecutionException;
+import org.elasticsearch.threadpool.Scheduler;
+import org.elasticsearch.threadpool.ThreadPool;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.function.Function;
+
+import static io.crate.replication.logical.repository.LogicalReplicationRepository.REMOTE_CLUSTER_REPO_REQ_TIMEOUT_IN_MILLI_SEC;
+
+public final class RemoteClusterStateMappingTracker implements Closeable {
+
+    private static final Logger LOGGER = Loggers.getLogger(RemoteClusterStateMappingTracker.class);
+
+    private final ThreadPool threadPool;
+    private final Function<String, Client> remoteClient;
+    private final ClusterService clusterService;
+    private final Set<String> subscriptionsToTrack = Collections.synchronizedSet(new HashSet<>());
+    private final TimeValue pollDelay;
+    private Iterator<TimeValue> delay;
+    private Scheduler.Cancellable cancellable;
+    private boolean isStarted = false;
+
+    public RemoteClusterStateMappingTracker(Settings settings, ThreadPool threadPool, Function<String, Client> remoteClient, ClusterService clusterService) {
```

Comment:
I think we can probably generalize `ShardReplicationChangesTracker` and `RemoteClusterStateMappingTracker` into one class containing all the Scheduling logic.